### PR TITLE
enhancement/AY-6116-blender_allow_handles_frame_range

### PR DESCRIPTION
**Pull request by jrsndl**
## Changelog Description

Allows to configure in settings whether a workfile frame range should include the handles or not for certain tasks. The default is like before where handles are excluded from the timeline.

## Additional info

Should address the issue #27 

Matches a similar feature in `ayon-maya`.

## Testing notes:

1. Upload new addon
2. Tweak the settings for Including handles or not
3. Trigger set frame range

### CODE REVIEW

**Comment by BigRoy** [comment]
Resolved the conflicts, pushed some cosmetic fixes and updated the PR description and added reviewers. Happy testing @LiborBatek 🫶

**Review by moonyuet — COMMENTED** [review]
When I enable the `include_handles_default`, the setting works as expected. But setting up the handle does not work when the task type being enabled,

When the `include_handles_default` settings is disabled, and with the tasktype filtering system set up, all the handles in all task types are not set up by clicking the set frame range.

**Review by LiborBatek — APPROVED** [review]
Been tested in both settings ```with handles``` used and when disabled and not used...

image


Works like charm!

**Comment by BigRoy** [comment]
@LiborBatek how about the case that Kayla mentioned:

> When I enable the include_handles_default, the setting works as expected. But setting up the handle does not work when the task type being enabled,
> 
> When the include_handles_default settings is disabled, and with the tasktype filtering system set up, all the handles in all task types are not set up by clicking the set frame range.

It sounds like @moonyuet says the task type filtering does not work as intended, or?

**Comment by BigRoy** [comment]
I had a look at the [Maya implementation here](https://github.com/ynput/ayon-maya/blob/fc52bb1eb3c1d5c5e581f781e76a2aa5871d8634/client/ayon_maya/api/lib.py#L2530) and I think I fixed it with https://github.com/ynput/ayon-blender/pull/28/commits/0d7d2e5190721aa46912e9e1e24fe0864850f0ca @moonyuet could you confirm?

File: server/settings/include_handles.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
+from ayon_server.settings import (
+    BaseSettingsModel,
+    SettingsField,
+    task_types_enum,
+)
+
+
+class IncludeByTaskTypeModel(BaseSettingsModel):
+    task_type: list[str] = SettingsField(
```
Comment (iLLiCiTiT) [inline server/settings/include_handles.py:9]:
```suggestion
    task_types: list[str] = SettingsField(
```

File: server/settings/include_handles.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        enum_resolver=task_types_enum
+    )
+    include_handles: bool = SettingsField(True, title="Include handles")
+
+
+class IncludeHandlesModel(BaseSettingsModel):
+    include_handles_default: bool = SettingsField(
+        True, title="Include handles by default"
+    )
+    per_task_type: list[IncludeByTaskTypeModel] = SettingsField(
```
Comment (iLLiCiTiT) [inline server/settings/include_handles.py:21]: Couldn't this be profiles?

File: server/settings/include_handles.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
+from ayon_server.settings import (
```
Comment (iLLiCiTiT) [inline server/settings/include_handles.py:1]: Is it so important it deservers separate file? 🙂

File: server/settings/include_handles.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+        default_factory=list,
+        title="Task types",
+        enum_resolver=task_types_enum
+    )
+    include_handles: bool = SettingsField(True, title="Include handles")
+
+
+class IncludeHandlesModel(BaseSettingsModel):
+    include_handles_default: bool = SettingsField(
+        True, title="Include handles by default"
```
Comment (iLLiCiTiT) [inline server/settings/include_handles.py:19]:
```suggestion
        False, title="Include handles by default"
```

File: server/settings/include_handles.py (hunk 2 above)
Comment (BigRoy) [reply server/settings/include_handles.py]: Done with https://github.com/ynput/ayon-blender/pull/28/commits/6534f1c5f390359f36ec3ed1bb8e0b0560ce4855

**Review by LiborBatek — CHANGES_REQUESTED** [review]
Yeah  I have made test before the ```Task Profiles``` and it was working fine as Kayla mentioned...

but when ```Profiles``` used it does not take handles into account, need to be addressed

**Review by LiborBatek — APPROVED** [review]
I have re-test it again and it works well now!
thanks goes to @iLLiCiTiT pushing (me) to redo it

**Pull request merged**